### DISCUSSION TITLE
Renovate not creating any PRs as it won't start enough jobs

### DISCUSSION THREAD

**Discussion by mlafeldt** [renovatebot/renovate#16893] [Request Help]
Hey,

We've been using Renovate for months and generally, it's been working great for us. 👍

However, for some reason, Renovate won't open PRs for our repo anymore.

I checked the logs at https://app.renovatebot.com/dashboard#github/grafbase/api, which indicate that, while there's currently exactly [one job](https://app.renovatebot.com/dashboard#github/grafbase/api/758323508), that one skipped creation of PRs as the job was started _after_ our configured schedule (weekly).

```
DEBUG: Checking schedule "before 3am on Monday"(branch="renovate-dependencies-non-major")
{
  "parsedSchedule": {
    "schedules": [
      {
        "t_b": [
          10800
        ],
        "d": [
          2
        ]
      }
    ],
    "exceptions": [],
    "error": -1
  }
}
DEBUG: Package not scheduled(branch="renovate-dependencies-non-major")
DEBUG: Skipping branch creation as not within schedule(branch="renovate-dependencies-non-major")
```

That's our config:

```json
{
  "branchPrefix": "renovate-",
  "extends": ["config:base", "schedule:weekly"],
  "packageRules": [
    {
      "groupName": "dependencies-non-major",
      "matchUpdateTypes": ["digest", "minor", "patch", "pin"]
    }
  ]
}
```

This morning, we changed the schedule to trick Renovate into opening PRs again:

```json
{
  "branchPrefix": "renovate-",
  "extends": ["config:base"],
  "schedule": ["after 8am on Monday"],
  "packageRules": [
    {
      "groupName": "dependencies-non-major",
      "matchUpdateTypes": ["digest", "minor", "patch", "pin"]
    }
  ]
}
```

However, that hasn't happened yet due to a lack of new jobs being fired.

In general, it seems that Renovate runs way too few jobs for the repo compared to other projects.

Using the Dependency Dashboard to trigger jobs manually didn't help either.

Please help. We really like Renovate and want to continue using it.

**Comment by mlafeldt** [comment]
The last PR Renovate created in our repo seems to be from June 27 (`https://github.com/grafbase/api/pull/878`).

**Comment by PhilipAbed** [comment]
according to the logs you posted, there's a problem in parsing the schedule you set, although i see something similar in our docs but it's not being parsed by the online parser

see [documentation ](https://docs.renovatebot.com/noise-reduction/#selective-scheduling) it says it's using `later.js` , 
i checked an online `later.js` parser: https://codepen.io/antonlydike/full/Kgygpr

parser [documentation](https://bunkat.github.io/later/parsers.html#text): 

try to find some schedule that suits you and that can get parsed with https://codepen.io/antonlydike/full/Kgygpr
(note the timezone is a bit problematic) 
see if you still get that error

**Reply by mlafeldt** [reply]
Hm, but we had been using `"extends": ["config:base", "schedule:weekly"]` for months until something broke as reported.

We're only trying different schedule dates now to get Renovate to work again.

**Reply by rarkins** [reply]
Renovate should currently be running on schedule every 2-3 hours per repo. It can run much more often (there is no limit) if there's activity in the repo, such as merged PRs or users requesting a run via checkboxes.

It's possible that no runs could happen on a repo during midnight-3am if the job queue gets particularly long at that time. Can you use the dashboard to confirm what the last job was before midnight and the first one was after 3am?

**Reply by mlafeldt** [reply]
Hey @rarkins,

Here are the recent jobs (not that many):

<img width="722" alt="Screen Shot 2022-08-03 at 10 40 50" src="https://user-images.githubusercontent.com/158074/182564503-bd6e382a-cd4b-42d6-adae-22260cf41cc1.png">

After tricking Renovate into opening PRs again by changing the weekly schedule to `after 8am`, we now see more errors like this one from `chore(deps): update dependencies-non-major`

```
WARN: artifactErrors(branch="renovate-dependencies-non-major")
{
  "artifactErrors": [
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    },
    {
      "lockFile": "Cargo.lock",
      "stderr": "Process exited with exit code \"101\""
    }
  ]
}
```

I don't think that's related to seeing too few jobs though.

Also noteworthy: while `Check this box to trigger a request for Renovate to run again on this repository` did tricker a job from the dashboard, clicking `If you want to rebase/retry this PR, click this checkbox.` inside PRs does nothing at all.

**Reply by PhilipAbed** [reply]
so you are saying the schedule:weekly isnt working as expected? can u send full debug log?

**Reply by mlafeldt** [reply]
Unfortunately, I can't send you logs for jobs that aren't executed. That's the entire problem. 😔

As reported, Renovate did start jobs, but they had missed the weekly schedule for weeks. See my first message for logs.

**Reply by PhilipAbed** [reply]
another question: 
you said on Other repositories it works fine right? so there might be something in this specific repository that is causing the issue

something happened at the 25/7 
![image](https://user-images.githubusercontent.com/42116482/182579276-39267cae-e3e6-49d6-a590-30aa1a8d4427.png)

jobs ran but they didnt finish? this happened until 31/7 jobs didnt end, (yet u pinged that a PR was opened at 27) 

i'm sure the logs of the last job (25/7) might help and if there is anything between 26-31/7 would be helpful too 
thing is those links you sent above are only available to you, i cant see them as a contributor.
i dont have access to `https://github.com/grafbase/api/pull/878` either looks like a private repository,

so either create us a minimal reproduction public repository to check it on, or show us some logs, 
nothing is helping get enough info to understand the issue for now.

**Reply by mlafeldt** [reply]
Unfortunately, the job log screenshot I posted above is complete. The last available logs are from 2 days ago, so no chance of looking at older logs. :/

I actually tried Renovate with a copy of the repo, which didn't work either. So you're right in that it's related to the repo's content somehow.

**Reply by PhilipAbed** [reply]
@mlafeldt can you create a public repo with only the package files ? like `cargo.toml` + `cargo.lock` +`renovate.json` ?
if there is anything sensitive remove it..

and try to run on it, if it reproduces, please show us that repository so we can fork it and investigate further
